core/feed_protocol/acquisition.py

```diff
@@ -0,0 +1,931 @@
+        works = patron.works_on_loan_or_on_hold()
+
+        feed = OPDSAcquisitionFeed("Active loans and holds", url, works, annotator)
+        feed.generate_feed()
+        response = feed.as_response(max_age=0, private=True)
+
+        last_modified = patron.last_loan_activity_sync
+        if last_modified:
+            response.last_modified = last_modified
+        return response
+
+    @classmethod
+    def single_entry_loans_feed(
+        cls,
+        circulation: Any,
+        item: LicensePool | Loan,
+        annotator: LibraryAnnotator | None = None,
+        fulfillment: FulfillmentInfo | None = None,
+        **response_kwargs: Any,
+    ) -> OPDSEntryResponse | ProblemDetail | None:
+        """A single entry as a standalone feed specific to a patron"""
+        if not item:
+            raise ValueError("Argument 'item' must be non-empty")
+
+        if isinstance(item, LicensePool):
+            license_pool = item
+            library = circulation.library
+        elif isinstance(item, (Loan, Hold)):
+            license_pool = item.license_pool
+            library = item.library
+        else:
+            raise ValueError(
+                "Argument 'item' must be an instance of {}, {}, or {} classes".format(
+                    Loan, Hold, LicensePool
+                )
+            )
+
+        if not annotator:
+            annotator = LibraryLoanAndHoldAnnotator(circulation, None, library)
+
+        log = logging.getLogger(cls.__name__)
+
+        # Sometimes the pool or work may be None
+        # In those cases we have to protect against the exceptions
+        try:
+            work = license_pool.work or license_pool.presentation_edition.work
+        except AttributeError as ex:
+            log.error(f"Error retrieving a Work Object {ex}")
+            log.error(
+                f"Error Data: {license_pool} | {license_pool and license_pool.presentation_edition}"
+            )
+            return NOT_FOUND_ON_REMOTE
+
+        if not work:
+            return NOT_FOUND_ON_REMOTE
+
+        _db = Session.object_session(item)
+        active_loans_by_work: Any = {}
+        active_holds_by_work: Any = {}
+        active_fulfillments_by_work = {}
+        item_dictionary = None
+
+        if isinstance(item, Loan):
+            item_dictionary = active_loans_by_work
+        elif isinstance(item, Hold):
+            item_dictionary = active_holds_by_work
+
+        if item_dictionary is not None:
+            item_dictionary[work] = item
+
+        if fulfillment:
+            active_fulfillments_by_work[work] = fulfillment
+
+        annotator.active_loans_by_work = active_loans_by_work
+        annotator.active_holds_by_work = active_holds_by_work
+        annotator.active_fulfillments_by_work = active_fulfillments_by_work
+        identifier = license_pool.identifier
+
+        entry = cls.single_entry(work, annotator, even_if_no_license_pool=True)
+
+        # TODO: max_age and private response kwargs
```

Comment:
Should this todo get sorted before merging?

Reply:
This is handled in #1340 since all these methods return feed objects, and the max_age and private options are handled within the `as_response` method.